Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -29,23 +29,44 @@ open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate, URLSessionDat
     }
 
     var instrumentation: URLSessionAutoInstrumentation? {
-        core?.v1.feature(URLSessionAutoInstrumentation.self)
+        coreReference.core?.v1.feature(URLSessionAutoInstrumentation.self)
     }
 
     let firstPartyHosts: FirstPartyHosts
 
-    /// The instance of the SDK core notified by this delegate.
-    /// It must be a weak reference, because `URLSessionDelegate` can last longer than this instance.
-    private weak var core: DatadogCoreProtocol?
+    private let coreReference: CoreReference
+
+    /// A wrapper for `DatadogCoreProtocol` reference to the SDK core notified by this delegate.
+    ///
+    /// It cannot be a strong reference, because `URLSessionDelegate` object can live longer than SDK instance.
+    enum CoreReference {
+        struct WeakReference {
+            weak var core: DatadogCoreProtocol?
+        }
+
+        /// No `core` was provided, use `defaultDatadogCore`.
+        case useDefaultCore
+        /// Some `core` object was provided, keep weak reference to it.
+        case useExplicitCore(WeakReference)
+
+        var core: DatadogCoreProtocol? {
+            switch self {
+            case .useDefaultCore:
+                return defaultDatadogCore
+            case .useExplicitCore(let reference):
+                return reference.core
+            }
+        }
+    }
```

Comment:
It make sense to either keep a weak ref to the provided core, if any, or lazy load the default one 👍 

I think we could simplify this with the following:
```swift
    var instrumentation: URLSessionAutoInstrumentation? {
        let core = core ?? defaultDatadogCore // lazy load default if no core.
        return core.v1.feature(URLSessionAutoInstrumentation.self)
    }

    let firstPartyHosts: FirstPartyHosts

    private weak var core: DatadogCoreProtocol?

    @objc
    override public init() {
        core = nil
        firstPartyHosts = .init()
        super.init()
    }

    public convenience init(in core: DatadogCoreProtocol? = nil) {
        self.init(in: core, additionalFirstPartyHostsWithHeaderTypes: [:])
    }

    @objc
    public convenience init(additionalFirstPartyHosts: Set<String>) {
        self.init(additionalFirstPartyHostsWithHeaderTypes: additionalFirstPartyHosts.reduce(into: [:], { partialResult, host in
            partialResult[host] = [.datadog]
        }))
    }

    public init(
        in core: DatadogCoreProtocol? = nil,
        additionalFirstPartyHostsWithHeaderTypes: [String: Set<TracingHeaderType>]
    ) {
        self.core = core
        self.firstPartyHosts = FirstPartyHosts(additionalFirstPartyHostsWithHeaderTypes)
        super.init()
    }
```

Reply:
🤦 Definitely 💯! 100x Much easier - thanks 😅